meta-resin-common/recipes-core/os-release/os-release.bbappend

```diff
@@ -1,9 +1,12 @@
 # Add custom resin fields
-OS_RELEASE_FIELDS_append = " RESIN_BOARD_REV META_RESIN_REV SLUG MACHINE"
+OS_RELEASE_FIELDS_append = " RESIN_BOARD_REV META_RESIN_REV SLUG MACHINE VARIANT VARIANT_ID"
 
 # Simplify VERSION output
 VERSION = "${HOSTOS_VERSION}"
 
+VARIANT = "${@bb.utils.contains('DEVELOPMENT_IMAGE','1','Development','Production',d)}"
+VARIANT_ID = "${@bb.utils.contains('DEVELOPMENT_IMAGE','1','dev','prod',d)}"
```

Comment:
@floion I believe the linked issue has all the details. 
https://github.com/resin-os/resinos/issues/173
https://www.freedesktop.org/software/systemd/man/os-release.html#VARIANT=